### ISSUE TITLE
Modules that implement Protocols fail to check when a tuple type member

### ISSUE THREAD

**Issue by tsibley** [microsoft/pyright#9988]
**Describe the bug**
A type of `tuple[Protocol]` fails to check when the protocol is implemented by a module. The module checks ok against the protocol when used outside of a tuple.

**Code or Screenshots**

```python
# x.py 
from typing import Protocol
from . import y

class YModule(Protocol):
    def f(self) -> str: ...

x1: YModule = y            # ok
x3: tuple[YModule] = (y,)  # fails
```

```python
# y.py
def f() -> str:
    return "plugh"
```

```console
$ npx pyright 
/tmp/tmp.6Rvu8doje4/x.py
  /tmp/tmp.6Rvu8doje4/x.py:9:22 - error: Type "tuple[ModuleType]" is not assignable to declared type "tuple[YModule]"
    "tuple[ModuleType]" is not assignable to "tuple[YModule]"
      Tuple entry 1 is incorrect type
        "ModuleType" is incompatible with protocol "YModule"
          "f" is not present (reportAssignmentType)
1 error, 0 warnings, 0 informations 
```

**VS Code extension or command-line**

```console
$ npx pyright --version
pyright 1.1.395
```

**Comment by erictraut** [comment]
Pyright doesn't currently support bidirectional type inference for modules, so it falls back on normal inference rules in this case. The default inference rules evaluate the type of `(y,)` as `tuple[ModuleType]`.

I'm going to recast this from a bug report to an enhancement request.

Mypy supports this case, so I think pyright should as well.

**Comment by tsibley** [comment]
Thanks, Eric. Is the best workaround at the moment using `cast()`? e.g.

```python
x: tuple[YModule] = (cast(YModule, y),)
```

I thought that I might be able to do this

```python
x1: YModule = y             # ok
x3: tuple[YModule] = (x1,)  # fails
```

but it also fails.

**Comment by erictraut** [comment]
This will be addressed in the next release.

Yeah, in the meantime, a `cast` is probably the best workaround.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.396.
